ddtrace/opentelemetry/span.go

```diff
@@ -0,0 +1,118 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package opentelemetry
+
+import (
+	"encoding/binary"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	otelcodes "go.opentelemetry.io/otel/codes"
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+var _ oteltrace.Span = (*span)(nil)
+
+type span struct {
+	ddtrace.Span
+	finished bool
+	statusInfo
+	*oteltracer
+}
+
+func (s *span) TracerProvider() oteltrace.TracerProvider                { return s.oteltracer.provider }
+func (s *span) AddEvent(name string, options ...oteltrace.EventOption)  { /*	no-op */ }
+func (s *span) RecordError(err error, options ...oteltrace.EventOption) { /*	no-op */ }
+
+func (s *span) SetName(name string) { s.SetOperationName(name) }
+
+func (s *span) End(options ...oteltrace.SpanEndOption) {
+	var finishCfg = oteltrace.NewSpanEndConfig(options...)
+	var localOpts []tracer.FinishOption
+	if s.statusInfo.code == otelcodes.Error {
+		s.SetTag(ext.ErrorMsg, s.statusInfo.description)
+	}
+	if t := finishCfg.Timestamp(); !t.IsZero() {
+		localOpts = append(localOpts, tracer.FinishTime(t))
+	}
+	s.Finish(localOpts...)
+	s.finished = true
```

Comment:
I imagine there wouldn't be a race condition here, but juuuust in case, perhaps set `s.finished = true` before `s.Finish`